app/core/Engine.js

```diff
@@ -64,70 +64,132 @@ class Engine {
+						setSelectedAddress: preferencesController.setSelectedAddress.bind(preferencesController)
+					},
+					{ encryptor },
+					initialState.KeyringController
+				),
+				new AccountTrackerController({
+					onPreferencesStateChange: listener => preferencesController.subscribe(listener),
+					getIdentities: () => preferencesController.state.identities
+				}),
+				new AddressBookController(),
+				assetsContractController,
+				assetsController,
+				new AssetsDetectionController({
+					onAssetsStateChange: listener => assetsController.subscribe(listener),
+					onPreferencesStateChange: listener => preferencesController.subscribe(listener),
+					onNetworkStateChange: listener => networkController.subscribe(listener),
+					getOpenSeaApiKey: () => assetsController.openSeaApiKey,
+					getBalancesInSingleCall: assetsContractController.getBalancesInSingleCall.bind(
+						assetsContractController
 					),
-					new TokenBalancesController({ interval: 10000 }),
-					new TokenRatesController(),
-					new TransactionController(),
-					new TypedMessageManager(),
-					new SwapsController({ clientId: AppConstants.SWAPS.CLIENT_ID })
-				],
-				initialState
-			);
+					addTokens: assetsController.addTokens.bind(assetsController),
+					addCollectible: assetsController.addCollectible.bind(assetsController),
+					removeCollectible: assetsController.removeCollectible.bind(assetsController),
+					getAssetsState: () => assetsController.state
+				}),
+				currencyRateController,
+				new PersonalMessageManager(),
+				new MessageManager(),
+				networkController,
+				new PhishingController(),
+				preferencesController,
+				new TokenBalancesController(
+					{
+						onAssetsStateChange: listener => assetsController.subscribe(listener),
+						getSelectedAddress: () => preferencesController.state.selectedAddress,
+						getBalanceOf: assetsContractController.getBalanceOf.bind(assetsContractController)
+					},
+					{ interval: 10000 }
+				),
+				new TokenRatesController({
+					onAssetsStateChange: listener => assetsController.subscribe(listener),
+					onCurrencyRateStateChange: listener => currencyRateController.subscribe(listener)
+				}),
+				new TransactionController({
+					getNetworkState: () => networkController.state,
+					onNetworkStateChange: listener => networkController.subscribe(listener),
+					getProvider: () => networkController.provider
+				}),
+				new TypedMessageManager(),
+				new SwapsController({ clientId: AppConstants.SWAPS.CLIENT_ID })
+			];
+
+			// set initial state
+			// TODO: Pass initial state into each controller constructor instead
+			// This is being set post-construction for now to ensure it's functionally equivalent with
+			// how the `ComponsedController` used to set initial state.
+			for (const controller of controllers) {
+				if (initialState[controller.name]) {
+					controller.update(this.cachedState[controller.name]);
```

Comment:
Oops! `cachedState` doesn't exist. That's what the `initialState` was renamed to inside the old `ComposedController` implementation.

```suggestion
					controller.update(initialState[controller.name]);
```